### PR TITLE
Base Miner Code Refactor

### PR DESCRIPTION
**Pull request by dylanuys**
- Renamed NPR-DeepfakeDetection to base_miner (while thanking authors in the readme ofc)
- Cleaned up training script, moved util functions to their own python modules
- Renamed RealImageDataset to ImageDataset, since we use this for fake images too now
- Made edits to the readme, markedup version available for review [here](https://docs.google.com/document/d/1mCZ8FaH7vXxwysMpe8u1qJ8uYEbNvh1Wh_BVHI12Tmw/edit)

### CODE REVIEW

File: base_miner/NPR.pth
Comment (kenobijon) [inline base_miner/NPR.pth:1]: [Suggestion] Keep it base_miner, more standard with other subnets

File: base_miner/options/__init__.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from .train_options import TrainOptions
+
+
+def get_val_opt():
```
Comment (kenobijon) [inline base_miner/options/__init__.py:4]: where is this used?

File: base_miner/util/data.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+from bitmind.image_dataset import ImageDataset
+from bitmind.data_sources import DATASET_META
+
+
+def load_datasets():
+    """
+    
+    Returns:
+
+    """
```
Comment (kenobijon) [inline base_miner/util/data.py:12]: Suggestion: Refactor comments for these, add brief description and remove spaces

File: base_miner/util/data.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+    Returns:
+
+    """
+    def CenterCrop():
+        def fn(img):
+            m = min(img.size)
+            return transforms.CenterCrop(m)(img)
+
+        return fn
```
Comment (kenobijon) [inline base_miner/util/data.py:47]: [Suggestion] Move this into utils or outside of function.

File: bitmind/miner/predict.py [hunk 4]
```diff
@@ -7,15 +7,14 @@ def fn(img):
         return transforms.CenterCrop(m)(img)
     return fn
 
+
+# these transforms should match the tranforms used during your model's training,
+# sans any data augmentation
 transform = transforms.Compose([
     CenterCrop(),
```
Comment (kenobijon) [inline bitmind/miner/predict.py:14]: See above, can move CenterCrop into a util file for re-use

File: neurons/validator.py [hunk 5]
```diff
@@ -42,23 +39,15 @@ class Validator(BaseValidatorNeuron):
...
-            {'name': 'jlbaker361/flickr_humans_20k', 'create_splits': True},
-            {'name': 'saitsharipov/CelebA-HQ', 'create_splits': True}
-        ]
         self.real_image_datasets = [
-            RealImageDataset(ds['name'], 'test', ds['create_splits']) for ds in real_image_dataset_meta
+            ImageDataset(ds['name'], 'test', ds['create_splits']) for ds in DATASET_META['real']
         ]
 
-        self.random_image_generator = RandomImageGenerator(diffuser_name='random')
+        self.random_image_generator = RandomImageGenerator(use_random_diffuser=True, diffuser_name=None)
```
Comment (kenobijon) [inline neurons/validator.py:50]: [Question] In what case would diffuser_name be used?

File: base_miner/NPR.pth
Comment (dylanuys) [reply base_miner/NPR.pth:1]: ah these are the weights given by the authors, trained on their data. I'll move it to the mining_models dir and rename

File: base_miner/options/__init__.py (hunk 1 above)
Comment (dylanuys) [reply base_miner/options/__init__.py:4]: it's not anymore. will remove.

**Review by kenobijon — APPROVED** [review]
This is good to go -- some future small refactors are suggested

File: base_miner/util/data.py (hunk 2 above)
Comment (dylanuys) [reply base_miner/util/data.py:12]: caught red handed not filling out my function headers

File: neurons/validator.py (hunk 5 above)
Comment (dylanuys) [reply neurons/validator.py:50]: `diffuser_name` would be used if the `RandomImageGenerator `was being used with just one specific diffuser.
it's also used as an attribute to check what diffuser the last image generated by the `RandomImageGenerator` came from (useful when `use_random_diffuser` is True)

Comment (dylanuys) [reply neurons/validator.py:50]: no plans to have validators set diffuser name, but I use it when generating datasets from a specific diffuser

**Pull request merged**